### ISSUE TITLE
Integrate issue tracker in-game for features, tweaks and bug reports

### ISSUE THREAD

**Issue by Frooxius** [Yellow-Dog-Man/Resonite-Issues#965]
### Is your feature request related to a problem? Please describe.

Currently GitHub is the only way to report issues with the software and request features. This results in a bit of a barrier of entry, that can be intimidating to people.

### Describe the solution you'd like

Our plan is to eventually implement in-game UI that will integrate with out GitHub. This will make it easier to make requests directly in-game, without need to register for a GitHub account or use GitHub's interface.

This will provide some benefits for the reports too. For example we can make it very easy to include recent logs and other information with the issue and apply additional filtering/processing for required forms.

However there are certain downsides as well. With lowered barrier, we will likely end up with larger volume of issues and lower quality of reports, which will make the quality matter more. It means that more issues will not be directly addressed. However we can still use those for general signals of issues and problems people are facing, which can be overall beneficial. 

### Describe alternatives you've considered

We considered a number of things:
- Our own issue tracker system (this requires too much engineering)
- Adding separate web interface (this still requires user to go and make an account somewhere or use a website outside of the platform)
- Just stick only to GitHub and provide guides on how to make an issue

**Comment by Lixquid** [comment]
Migrating a concern over from the closed ticket #3137; whatever form error log reporting will encompass in the future, please bear in mind that some people will not want the information contained in the log files to be made completely public and would prefer them to only be shared with the developer team, due to privacy or other concerns.

**Comment by blhsrwznrghfzpr** [comment]
Froox previously mentioned that GitHub 👍 reactions are used to gauge community interest.
For reference: https://github.com/Yellow-Dog-Man/Resonite-Issues/discussions/5831#discussioncomment-14956101

However, many Resonite users are not programmers and are unfamiliar with GitHub, so very few people will create an account just to leave a 👍. One of my friends even told me they support the idea but *don’t want to create a GitHub account*, so they cannot up-vote at all.

Because of this, I would appreciate it if you could either:

* provide a way for users to express their feedback using their **Resonite account**, or
* reconsider relying on GitHub 👍 reactions as the primary measure of community interest.

**Comment by Frooxius** [comment]
We do plan to have in-game system for this. However developing our own system/integration takes some amount of time, compared to using existing solution (like GitHub) and it needs to be prioritized - meaning if we were to prioritize it now, we'd have to bump off other work and priorities.

So this brings question - should we work on this before rework UI? Before we rework IK? Before we add ProtoFlux collections? And other things? 

Would prioritizing this before those bring bigger benefit to growth and support for the platform over those issues?

**Comment by Redd56** [comment]
I think another thing to consider is survivorship bias.

What i mean specifically is that everyone upvoting stuff in github atleast somewhat plays the game. Never any brand new people who are leaving the game due to a high friction point. if there was a super low friction point way to upvote issues in game there could be a better way to gauge what needs to be prioritized for user retention. Rather than relying on the users already on the platform voicing their opinions.  

It also agree with what everyone else is saying, that being; due to not everyone that plays reso having or wanting to make a github or even visit the github (idk if theres any sort of stats for it anywhere for percentages, or how you would even collect them) that the vast majority of github issues upvoted are by the more technical players of the game. That is to say less casual people who just play for say 8 hours with friends for the purely social parts of the game instead of building. 

This leads to another sort of bias away from what the average player wants and more towards what certain demographics want. Same issue with other ways for the community to engage like discord, due to not everyone having a discord (though it is probably a majority) some users say ones that stay fully private dont get to voice their opinions in chats or polls that are conducted there. 


tldr: i actualy do think it should be one of the top priorities, even  above finishing the upgrade to the latest version of bepu even though i really really really wanna enable the particle systems i have strewn across my various creations but cant due to the infinite tree collider crash

**Comment by Frooxius** [comment]
This is actually not quite true, the upvotes are part of the decision making process, but not the whole thing. 

I strongly recommend redoing the "How we prioritize" document, which goes into detail on this: https://github.com/Yellow-Dog-Man/Resonite-Issues/blob/main/HOW_WE_PRIORITIZE.md

Everything considered, I'm not convinced this is a top priority for the platform right now, especially above the other issues.

It's not going to increase the usability and friendliness of the platform itself and I don't see it bringing a lot of new users (if any) and making more users stay.

It's also doing things out of order. To maket into an in-game UI, we'll need to rework more UI and build our in-game UI systems more anyways.

You mention that you feel that all those issues should be done "soon", but that's not how it really works.

Development takes time, you can't have everything at once - we have to pick which ones we do soon and which ones we'll do later.

If this one was done soon, then the other ones would have to be done later.

And right now I don't see strong value from doing this one sooner over the other issues.

**Comment by blhsrwznrghfzpr** [comment]
I understand that tasks need to be prioritized, and that certain foundational systems must be built first. Ultimately, YDMS will decide what should take priority, and I respect that.

My concern is simply that relying mainly on GitHub 👍 reactions can make it feel like the opinions of users who don’t use GitHub are not being represented. When someone posts on GitHub *on behalf of others*, it would be helpful if those signals could also be considered, rather than only the 👍 count.
